jgrapht-core/src/main/java/org/jgrapht/alg/matching/GreedyWeightedMatching.java

```diff
@@ -0,0 +1,176 @@
+ * weights and then greedily constructs a maximal matching using this ordering. A maximal matching
+ * (not to be confused with maximum) is a matching that cannot be increased in cardinality without
+ * removing an edge first.
+ *
+ * <p>
+ * For more information about approximation algorithm for the maximum weight matching problem in
+ * arbitrary graphs see:
+ * <ul>
+ * <li>R. Preis, Linear Time 1/2-Approximation Algorithm for Maximum Weighted Matching in General
+ * Graphs. Symposium on Theoretical Aspects of Computer Science, 259-269, 1999.</li>
+ * <li>D.E. Drake, S. Hougardy, A Simple Approximation Algorithm for the Weighted Matching Problem,
+ * Information Processing Letters 85, 211-213, 2003.</li>
+ * </ul>
+ * 
+ * @see PathGrowingWeightedMatching
+ * 
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ * 
+ * @author Dimitrios Michail
+ * @since September 2016
+ */
+public class GreedyWeightedMatching<V, E>
+    implements WeightedMatchingAlgorithm<V, E>
+{
+    /**
+     * Tolerance used when comparing floating point values.
+     */
+    public static final double DEFAULT_EPSILON = 1e-9;
+
+    private Set<E> matching;
+    private double weight;
+    private final double epsilon;
+
+    /**
+     * Create and execute a new instance of the greedy maximum weight matching algorithm. Floating
+     * point values are compared using {@link #DEFAULT_EPSILON} tolerance.
+     * 
+     * @param graph the input graph
+     */
+    public GreedyWeightedMatching(Graph<V, E> graph)
+    {
+        this(graph, DEFAULT_EPSILON);
+    }
+
+    /**
+     * Create and execute a new instance of the greedy maximum weight matching algorithm.
+     * 
+     * @param graph the input graph
+     * @param epsilon tolerance when comparing floating point values
+     */
+    public GreedyWeightedMatching(Graph<V, E> graph, double epsilon)
+    {
+        if (graph == null) {
+            throw new IllegalArgumentException("Input graph cannot be null");
+        }
+        if (epsilon < 0.0) {
+            throw new IllegalArgumentException("Tolerance cannot be negative");
+        }
+        this.epsilon = epsilon;
+
+        Pair<Double, Set<E>> result;
+        result = run(graph);
+
+        this.weight = result.first;
+        this.matching = result.second;
+    }
+
+    /**
+     * Get a matching that is a 1/2-approximation of the maximum weighted matching.
+     * 
+     * @return a matching
+     */
+    @Override
+    public Set<E> getMatching()
+    {
+        return Collections.unmodifiableSet(matching);
+    }
+
+    /**
+     * Get the weight of the computed matching.
+     * 
+     * @return the weight of the matching
+     */
+    @Override
+    public double getMatchingWeight()
+    {
+        return weight;
+    }
+
+    // the algorithm
+    private Pair<Double, Set<E>> run(Graph<V, E> graph)
+    {
+        // sort edges
+        ArrayList<E> allEdges = new ArrayList<>(graph.edgeSet());
+        Collections.sort(
```

Comment:
It does indeed, but the edge weights are reversed inside the lambda. Should I add a comment?